waspc/data/Generator/templates/server/src/auth/providers/oauth/init.ts

```diff
@@ -0,0 +1,79 @@
+import passport from "passport";
+
+import waspServerConfig from '../../../config.js';
+import { contextWithUserEntity } from "../../utils";
+
+import { InitData, ProviderConfig, OAuthConfig, RequestWithWasp } from "../types.js";
+
+export function makeOAuthInit({ userDefinedConfigFn, getUserFieldsFn, npmPackage, oAuthConfig }: OAuthImports) {
+    return async function init(provider: ProviderConfig): Promise<InitData> {
+        const userDefinedConfig = userDefinedConfigFn
+            ? await userDefinedConfigFn()
+            : {};
+        const ProviderStrategy = await import(npmPackage);
+
+        const passportStrategyName = `wasp${provider.slug}LoginStrategy`;
+        const requiredConfig = {
+            clientID: oAuthConfig.clientID,
+            clientSecret: oAuthConfig.clientSecret,
+            scope: oAuthConfig.scope,
+            callbackURL: `${waspServerConfig.frontendUrl}/auth/login/${provider.slug}`,
+            passReqToCallback: true
+        };
+
+        const config = {
+            ...requiredConfig,
+            ...userDefinedConfig,
```

Comment:
I wanted to enable the user to override the `scope`. If the user needs to override something, he's an advanced user, I would allow it.

We are calling `ensureValidConfig` after this, so he will get the error message if he's missing `clientId`, `clientSecret` or `scope` :)

Reply:
Ok, sounds good to me 👍🏻